visualization/dashboard/src/MLIDashboard/ValidateProperties.ts

```diff
@@ -0,0 +1,225 @@
+        if (props.probabilityY) {
+            if (!Array.isArray(props.probabilityY)) {
+                props.probabilityY = undefined;
+                if (rowLength !== undefined) {
+                    rowLength = 0;
+                }
+                return localization.formatString(localization.ValidationErrors.predictedProbNotArray, rowLength, classLength);
+            }
+            const length = props.probabilityY.length;
+            if (rowLength === undefined) {
+                rowLength = length;
+            }
+            if (length !== rowLength) {
+                return `Inconsistent dimensions. Predicted y has length [${length}], expected [${rowLength}]`;
+            }
+            if (length === 0) {
+                return "Predicted probability input not a non-empty array";
+            }
+            const cLength = props.probabilityY[0].length;
+            if (cLength !== classLength) {
+                return `Inconsistent dimensions. Predicted probability has dimensions [${length} x ${cLength}], expected [${rowLength} x ${classLength}]`;
+            }
+            if (!props.probabilityY.every(row => row.length === classLength)) {
+                return `Inconsistent dimensions. Predicted probability has rows of varying length`;
+            }
+        }
+    }
+
+    private validatePredictedY(): void {
+        const length = this.props.predictedY.length;
+        if (this.rowLength === undefined) {
+            this.rowLength = length;
+        }
+        if (length !== this.rowLength) {
+            this.props.predictedY = undefined;
+            this.errorStrings.push(localization.formatString(localization.ValidationErrors.inconsistentDimensions, localization.ValidationErrors.predictedY, length, this.rowLength));
+        }
+    }
+
+    private validatePredictProba(): void {
+        if (!Array.isArray(this.props.probabilityY)) {
+            this.props.probabilityY = undefined;
+            this.errorStrings.push(localization.formatString(localization.ValidationErrors.notArray, localization.ValidationErrors.predictedProbability, `${this.rowLength || 0} x ${this.classLength}`));
+            return;
+        }
+        const length = this.props.probabilityY.length;
+        if (this.rowLength === undefined) {
+            this.rowLength = length;
+        }
+        if (length !== this.rowLength) {
+            this.props.probabilityY = undefined;
+            this.errorStrings.push(localization.formatString(localization.ValidationErrors.inconsistentDimensions, localization.ValidationErrors.predictedProbability, length, this.rowLength))
+            return;
+        }
+        if (length === 0) {
+            this.props.probabilityY = undefined;
+            this.errorStrings.push(localization.formatString(localization.ValidationErrors.notNonEmpty, localization.ValidationErrors.predictedProbability))
+            return;
+        }
+        const cLength = this.props.probabilityY[0].length;
```

Comment:
nit: we seem to prefer the longer-form for the variable names in this repo
cLength => classLength